Lib/test/test_types.py

```diff
@@ -665,6 +665,11 @@ def test_or_type_operator_with_TypeVar(self):
         assert TV | str == typing.Union[TV, str]
         assert str | TV == typing.Union[str, TV]
 
+    def test_union_parameter_chaining(self):
+        T = typing.TypeVar("T")
+
+        self.assertEqual((float | list[T])[int], float | list[int])
+
```

Comment:
Oh can you add one more test in ``test_genericalias`` too please?
Specifically:
```suggestion
self.assertEqual(list[int | list[T]].__parameters__, (T,))
self.assertEqual(list[int | list[T]][str], list[int | list[str]])
```
The idea is to make sure GenericAlias always looks into nested Unions for ``__parameters__``, and nested unions inside genericaliases work.